core/src/main/scala/kafka/log/ProducerStateManager.scala

```diff
@@ -455,7 +459,18 @@ object ProducerStateManager {
 
   private def deleteSnapshotFiles(dir: File, predicate: Long => Boolean = _ => true) {
     listSnapshotFiles(dir).filter(file => predicate(offsetFromFile(file))).foreach { file =>
-      Files.deleteIfExists(file.toPath)
+      try{
+        logger.info(s"Sleep before deleteIfExists, thread Id: ${Thread.currentThread().getId}")
+        Thread.sleep(Thread.currentThread().getId % 200)
+        Files.deleteIfExists(file.toPath)
+      }
+      catch {
+        case e: IOException => {
+          logger.warn(s"Exception happened to deleteIfExists, error: ${e.getMessage}. Retrying. Sleep before deleteIfExists, thread Id: ${Thread.currentThread().getId}")
```

Comment:
I would prefer to calculate the sleep time ahead, and put the sleep time and the file name as part of the log message. Like, "Exception while deleting snapshot file '{file.toPath}'. will retry again after {milliSecs}

Reply:
sure